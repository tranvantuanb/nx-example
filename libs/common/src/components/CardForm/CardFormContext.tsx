import React, { useCallback, useMemo, useReducer, createContext } from 'react';
import { createAction, createReducer } from '@reduxjs/toolkit';

type CardFormMode = 'create' | 'edit';

interface CardFormContextValue {
  isEditing: boolean;
  mode: CardFormMode;
  // at the moment for simplicity, data sharing in the form context does not require type check
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  data: Record<string, any>;
}

const INITIAL_STATE: CardFormContextValue = {
  isEditing: false,
  mode: 'edit',
  data: {},
};

// ------------------------------------------- ACTIONS -------------------------------------------
const Action = {
  setEditing: createAction<boolean>('CARD_FORM/SET_EDITING'),
  setMode: createAction<CardFormMode>(`CARD_FORM/SET_MODE`),
  setData: createAction<Record<string, any>>(`CARD_FORM/SET_DATA`),
};

// ------------------------------------------- REDUCER -------------------------------------------
const reducer = createReducer(INITIAL_STATE, (builder) => {
  builder.addCase(Action.setEditing, (state, { payload }) => {
    state.isEditing = payload;
  });
  builder.addCase(Action.setMode, (state, { payload }) => {
    state.mode = payload;
  });
  builder.addCase(Action.setData, (state, { payload }) => {
    state.data = payload;
  });
});

export const CardFormContext = createContext({
  ...INITIAL_STATE,
  setEditing: (payload: boolean) => {},
  setMode: (payload: CardFormMode) => {},
  setData: (payload: Record<string, any>) => {},
});

export const CardFormContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);

  const setEditing = useCallback(
    (payload: boolean) => dispatch(Action.setEditing(payload)),
    []
  );

  const setMode = useCallback(
    (payload: CardFormMode) => dispatch(Action.setMode(payload)),
    []
  );

  const setData = useCallback(
    (payload: Record<string, any>) => dispatch(Action.setData(payload)),
    []
  );

  const value = useMemo(
    () => ({ ...state, setEditing, setMode, setData }),
    [state, setEditing, setMode, setData]
  );
  return (
    <CardFormContext.Provider value={value}>
      {children}
    </CardFormContext.Provider>
  );
};
