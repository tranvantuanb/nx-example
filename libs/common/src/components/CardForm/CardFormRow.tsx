import React from 'react';
import { Row, RowProps } from 'antd';

export type CardFormRowProps = RowProps;

export const CardFormRow: React.FC<CardFormRowProps> = React.memo(
  ({ children, ...rowProps }) => {
    return (
      <Row gutter={[16, 16]} {...rowProps}>
        {children}
      </Row>
    );
  }
);
