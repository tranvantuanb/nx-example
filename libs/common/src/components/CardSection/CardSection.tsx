import React from 'react';
import tw, { styled } from 'twin.macro';

import { Text } from '../Text/Text';

export interface CardSectionProps {
  className?: string;
  header?: React.ReactNode;
  children?: React.ReactNode;
  noMargin?: boolean;
}

const StyledCardSection = styled.div<Pick<CardSectionProps, 'noMargin'>>`
  ${({ noMargin }) => !noMargin && tw`mb-[32px]`}
`;

export const CardSection: React.FC<CardSectionProps> = React.memo(
  ({ className, header, noMargin, children, ...props }) => {
    return (
      <StyledCardSection noMargin={noMargin} className={className} {...props}>
        {!!header && (
          <Text.Caption size="sm" uppercase className="mb-[16px]">
            {header}
          </Text.Caption>
        )}
        {children}
      </StyledCardSection>
    );
  }
);
