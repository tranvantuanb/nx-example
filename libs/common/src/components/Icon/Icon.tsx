import React from 'react';
import { ReactSVG } from 'react-svg';
import { css, styled } from 'twin.macro';

export const IconNames = [
  'arrow-bottom-left',
  'arrow-down',
  'arrow-left',
  'arrow-top-right',
  'arrow-up',
  'attach',
  'calendar',
  'call',
  'check',
  'clock',
  'cross-circle',
  'car',
  'edit',
  'email',
  'exclamation',
  'location',
  'mail',
  'message',
  'one-way',
  'person',
  'question',
  'send',
  'ticket',
  'transfer',
  'two-way',
  'whatsapp',
  'whatsapp-color',
  'textbox',
  'close',
  'upload',
  'success',
  'location-pin',
  'three-dot-verticle',
  'enter',
  'eye',
  'filter',
  'delete',
  'plus-circle',
  'log-out',
  'auction',
  'seals',
  'wrench',
  'open-link',
] as const;

export type IconName = typeof IconNames[number];

interface StyledReactSVGProps {
  fill?: string;
  color?: string;
  size?: string;
  transform?: string;
  className?: string;
}

interface ComponentProps extends StyledReactSVGProps {
  name: IconName;
}

//@ts-ignore
const StyledSVGIcon = styled(ReactSVG)<StyledReactSVGProps>`
  svg {
    ${({ size }) =>
      size &&
      css`
        width: ${size};
        height: ${size};
      `}
    ${({ transform }) =>
      transform &&
      css`
        transform: ${transform};
      `}
    ${({ color }) =>
      color &&
      css`
        color: ${color};
      `}
    path {
      ${({ fill, color }) =>
        (fill || color) &&
        css`
          fill: ${fill || color};
        `}
    }
  }
`;

export const Icon: React.FC<ComponentProps> = React.memo(
  ({ name, color, fill, size, transform, className }) => {
    return (
      <StyledSVGIcon
        src={`/icons/${name}.svg`}
        color={color}
        fill={fill}
        size={size}
        transform={transform}
        className={className}
      />
    );
  }
);
