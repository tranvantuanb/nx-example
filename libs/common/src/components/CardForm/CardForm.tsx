import React, { useContext, useEffect } from 'react';
import { Form, FormProps } from 'antd';

import { CardFormContext, CardFormContextProvider } from './CardFormContext';
import {
  CardFormItem,
  CardFormItemProps as CardFormItemPropsInternal,
} from './CardFormItem/CardFormItem';

import {
  CardFormRow,
  CardFormRowProps as CardFormRowPropsInternal,
} from './CardFormRow';

import {
  CardFormActions,
  CardFormActionsProps as CardFormActionsPropsInternal,
} from './CardFormActions';

export interface CardFormProps {
  className?: string;
  /**
   * Children components should be CardForm.Item
   */
  children?: React.ReactNode;
  /**
   * Prop to control View/Editing mode
   */
  isEditing?: boolean;
  /**
   * Prop to indicate if the form is for newly creating
   */
  mode?: 'create' | 'edit';
}

export type CardFormItemProps = CardFormItemPropsInternal;
export type CardFormRowProps = CardFormRowPropsInternal;
export type CardFormActionsProps = CardFormActionsPropsInternal;
interface ICardForm extends React.FC<CardFormProps & FormProps> {
  Item: typeof CardFormItem;
  Row: typeof CardFormRow;
  Actions: typeof CardFormActions;
}

const CardForm: ICardForm = ({
  children,
  isEditing,
  mode,
  ...antFormProps
}) => {
  const { setEditing, setMode } = useContext(CardFormContext);

  useEffect(() => {
    setEditing(!!isEditing);
    setMode(mode || 'edit');
  }, [isEditing, mode, setEditing, setMode]);

  return (
    <Form layout="vertical" {...antFormProps}>
      {children}
    </Form>
  );
};

const withCardFormContext = <T,>(WrappedComponent) => {
  const Wrapper = React.memo((props: T) => (
    <CardFormContextProvider>
      <WrappedComponent {...props} />
    </CardFormContextProvider>
  ));
  return Wrapper;
};

CardForm.Item = CardFormItem;
CardForm.Row = CardFormRow;
CardForm.Actions = CardFormActions;

export { CardForm, withCardFormContext };
