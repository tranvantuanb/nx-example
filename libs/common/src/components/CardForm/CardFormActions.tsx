import { Form } from 'antd';
import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';

import { Button } from '../Button';
import { Space } from '../Space';
import { CardFormContext } from './CardFormContext';

export interface CardFormActionsProps {
  onCancel?: () => void;
  isSubmitting?: boolean;
}

export const CardFormActions: React.FC<CardFormActionsProps> = React.memo(
  ({ onCancel, isSubmitting }) => {
    const { t } = useTranslation('common');
    const { setEditing } = useContext(CardFormContext);
    const form = Form.useFormInstance();

    const handleCancel = () => {
      setEditing(false);
      form.resetFields();
      onCancel?.();
    };

    return (
      <Space>
        <Button type="text" onClick={handleCancel}>{t`Cancel`}</Button>
        <Button
          htmlType="submit"
          type="primary"
          loading={!!isSubmitting}
        >{t`Done`}</Button>
      </Space>
    );
  }
);
