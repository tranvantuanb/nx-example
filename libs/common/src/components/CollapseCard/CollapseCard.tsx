import React from 'react';
import {
  Collapse as AntCollapse,
  Divider as AntDivider,
  Space,
  SpaceProps,
} from 'antd';
import tw, { styled } from 'twin.macro';

import { Icon, IconName } from '../Icon';

export interface CollapseCardProps {
  className?: string;
  /**
   * Icon on the left of the header.
   */
  icon?: IconName | JSX.Element;
  /**
   * Header title
   */
  header: React.ReactNode;
  /**
   * Sub header title
   */
  subheader?: React.ReactNode;
  /**
   * Set the collapse open as default
   */
  defaultActive?: boolean;
  /**
   * Collapse card content
   */
  children?: React.ReactNode;
  /**
   * Extra items adding to the right of the header
   */
  extra?: React.ReactNode;
}
export interface CollapseCardEditIconProps {
  className?: string;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

interface CollapseCardInterface extends React.FC<CollapseCardProps> {
  Divider: typeof Divider;
  EditIcon: typeof EditIcon;
  FooterActions: typeof FooterActions;
}

const StyledCollapseCard = styled(AntCollapse)`
  ${tw`border-gray-200 bg-white`}
  .ant-collapse-item {
    ${tw`border-gray-200`}

    .ant-collapse-arrow {
      transition: all 0.3s ease-out;
      top: 24px !important;
      &.active {
        transform: translate(0, -6px) rotate(-180deg);
      }
    }
  }

  .ant-collapse-content {
    ${tw`border-t-0`}
    .ant-collapse-content-box {
      ${tw`pt-0`}
    }
  }

  form.ant-form {
    ${tw`mx-[8px]`}
  }
`;

const Divider = styled(AntDivider)`
  ${tw`my-[20px] border-gray-200`}
`;

const EditIcon: React.FC<CollapseCardEditIconProps> = ({
  className,
  onClick,
}) => (
  <button
    className={className}
    tw="absolute top-[-12px] right-[0px]"
    onClick={onClick}
  >
    <Icon name="edit" size="16px" />
  </button>
);

const FooterActions: React.FC<SpaceProps> = ({
  children,
  className,
  ...otherSpaceProps
}) => (
  <div className={className} tw="flex flex-row-reverse">
    <Space {...otherSpaceProps}>{children}</Space>
  </div>
);

const Header: React.FC<
  Pick<CollapseCardProps, 'icon' | 'header' | 'subheader'>
> = React.memo(({ icon, header, subheader }) => (
  <div className="header-wrapper flex flex-col">
    <div className="flex flex-row">
      {icon && (
        <div className="header-icon flex items-center mr-[12px]">
          {typeof icon === 'string' ? (
            <Icon name={icon as IconName} size="16px" color="black" />
          ) : (
            icon
          )}
        </div>
      )}
      <div className="header-content font-bold text-base">{header}</div>
    </div>
    {subheader && (
      <div className="subheader-content text-sm text-black-300 mt-[8px]">
        {subheader}
      </div>
    )}
  </div>
));

const Extra: React.FC<Pick<CollapseCardProps, 'extra'>> = React.memo(
  ({ extra }) =>
    !extra ? null : (
      <div
        onClick={(e) => {
          e.preventDefault();
          e.stopPropagation();
        }}
      >
        {extra}
      </div>
    )
);

const CollapseCard: CollapseCardInterface = ({
  className,
  icon,
  header,
  subheader,
  defaultActive = false,
  extra,
  children,
}) => {
  return (
    <StyledCollapseCard
      className={className}
      defaultActiveKey={defaultActive ? ['main-panel'] : undefined}
      expandIcon={({ isActive }) => (
        <Icon name="arrow-down" className={isActive ? 'active' : undefined} />
      )}
      expandIconPosition="end"
    >
      <AntCollapse.Panel
        header={<Header icon={icon} header={header} subheader={subheader} />}
        key="main-panel"
        extra={<Extra extra={extra} />}
      >
        <Divider className="mt-0" />
        <div className="relative">{children}</div>
      </AntCollapse.Panel>
    </StyledCollapseCard>
  );
};

CollapseCard.Divider = Divider;
CollapseCard.EditIcon = EditIcon;
CollapseCard.FooterActions = FooterActions;

export { CollapseCard };
