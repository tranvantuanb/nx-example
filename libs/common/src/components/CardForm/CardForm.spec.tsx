import React, { useContext, useEffect } from 'react';
import { Layout, Input } from 'antd';
import { render } from '../../test';

import { CollapseCard } from '../CollapseCard/CollapseCard';
import { CardSection } from '../CardSection/CardSection';
import { Icon } from '../Icon';
import { CardForm, withCardFormContext } from './CardForm';
import { CardFormContext } from './CardFormContext';

const initialValues = {
  contact: {
    name: 'Cristiano Ronaldo',
    contact_number: '77777777',
    email: 'manchester_united_are_rubbish@bbc.com',
  },
  slug: 'QWERTY',
  listing: {
    id: 12345,
  },
};

const CollapseCardPage = withCardFormContext(() => {
  const { isEditing, setData, setEditing } = useContext(CardFormContext);
  useEffect(() => {
    setData(initialValues);
  }, [setData]);

  const handleFinish = (formValues) => {
    console.log('formValues', formValues);
  };

  return (
    <Layout>
      <Layout.Content>
        <div className="container my-0 mx-auto p-14 bg-white">
          <CollapseCard
            header="Customer Input"
            icon={<Icon name="person" color="black" />}
            defaultActive
          >
            <CollapseCard.EditIcon onClick={() => setEditing(!isEditing)} />
            <CardForm initialValues={initialValues} onFinish={handleFinish}>
              <CardSection header="Contact Details">
                <CardForm.Row>
                  <CardForm.Item name={['contact', 'name']} label="Name">
                    <Input />
                  </CardForm.Item>
                  <CardForm.Item
                    name={['contact', 'contact_number']}
                    label="Contact Number"
                    renderViewValue={(number) =>
                      number.match(/.{1,4}/g).join(' ')
                    }
                  >
                    <Input />
                  </CardForm.Item>
                  <CardForm.Item name={['contact', 'email']} label="Email">
                    <Input />
                  </CardForm.Item>
                </CardForm.Row>
              </CardSection>
              <CardSection header="Ticket Data">
                <CardForm.Row>
                  <CardForm.Item name="slug" label="Slug">
                    <Input />
                  </CardForm.Item>
                  <CardForm.Item name={['listing', 'id']} label="Listing ID">
                    <Input />
                  </CardForm.Item>
                </CardForm.Row>
              </CardSection>
              {isEditing && (
                <CollapseCard.FooterActions>
                  <CardForm.Actions />
                </CollapseCard.FooterActions>
              )}
            </CardForm>
          </CollapseCard>
        </div>
      </Layout.Content>
    </Layout>
  );
});

describe('CardForm', () => {
  it('should render successfully', () => {
    const { container } = render(<CollapseCardPage />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
