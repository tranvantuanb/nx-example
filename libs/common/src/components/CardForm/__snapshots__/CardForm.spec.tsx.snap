// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardForm should render successfully 1`] = `
.c5 {
  margin-top: 0px;
}

.c7 {
  position: absolute;
  top: -12px;
  right: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 12px;
}

.c4 {
  font-weight: 700;
  font-size: 16px;
}

.c6 {
  position: relative;
}

.c8 {
  display: block;
}

.c9 {
  margin-bottom: 16px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  font-size: 14px;
  line-height: 14px;
  font-weight: 700;
  --tw-text-opacity: 1;
  color: rgba(161,167,188,var(--tw-text-opacity));
}

.c12 {
  font-size: 14px;
  line-height: 14px;
  font-weight: 700;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 3.5rem;
  --tw-bg-opacity: 1;
  background-color: rgba(255,255,255,var(--tw-bg-opacity));
}

@media (min-width:376px) {
  .c0 {
    max-width: 376px;
  }
}

@media (min-width:480px) {
  .c0 {
    max-width: 480px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 768px;
  }
}

@media (min-width:976px) {
  .c0 {
    max-width: 976px;
  }
}

@media (min-width:1280px) {
  .c0 {
    max-width: 1280px;
  }
}

@media (min-width:1536px) {
  .c0 {
    max-width: 1536px;
  }
}

@media (min-width:1800px) {
  .c0 {
    max-width: 1800px;
  }
}

<div>
  <section
    class="ant-layout"
  >
    <main
      class="ant-layout-content"
    >
      <div
        class="c0"
      >
        <div
          class="ant-collapse ant-collapse-icon-position-end css-pkcp1s-StyledCollapseCard eyl3bx71"
        >
          <div
            class="ant-collapse-item ant-collapse-item-active"
          >
            <div
              aria-expanded="true"
              class="ant-collapse-header"
              role="button"
              tabindex="0"
            >
              <div
                class="ant-collapse-expand-icon"
              >
                <div
                  class="active ant-collapse-arrow css-pwgt72-StyledSVGIcon e81aohe0"
                >
                  <div>
                    <div
                      data-src="/icons/arrow-down.svg"
                      src=""
                    />
                  </div>
                </div>
              </div>
              <span
                class="ant-collapse-header-text"
              >
                <div
                  class="c1 header-wrapper"
                >
                  <div
                    class="c2"
                  >
                    <div
                      class="c3 header-icon"
                    >
                      <div
                        class="css-10v150v-StyledSVGIcon-StyledSVGIcon-StyledSVGIcon e81aohe0"
                        color="black"
                      >
                        <div>
                          <div
                            data-src="/icons/person.svg"
                            src=""
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="c4 header-content"
                    >
                      Customer Input
                    </div>
                  </div>
                </div>
              </span>
              <div
                class="ant-collapse-extra"
              />
            </div>
            <div
              class="ant-collapse-content ant-collapse-content-active"
            >
              <div
                class="ant-collapse-content-box"
              >
                <div
                  class="ant-divider ant-divider-horizontal c5 css-5qn4iy-Divider eyl3bx70"
                  role="separator"
                />
                <div
                  class="c6"
                >
                  <button
                    class="c7"
                  >
                    <div
                      class="css-na5dur-StyledSVGIcon-StyledSVGIcon e81aohe0"
                    >
                      <div>
                        <div
                          data-src="/icons/edit.svg"
                          src=""
                        />
                      </div>
                    </div>
                  </button>
                  <form
                    class="ant-form ant-form-vertical"
                  >
                    <div
                      class="css-1ijzjjj-StyledCardSection e5y6tk60"
                    >
                      <span
                        class="ant-typography c8 c9 css-14xdwh8-StyledText-StyledTextCaption emffahp0"
                      >
                        <strong>
                          Contact Details
                        </strong>
                      </span>
                      <div
                        class="ant-row"
                        style="margin: -8px -8px -8px -8px;"
                      >
                        <div
                          class="ant-col ant-col-24"
                          style="padding: 8px 8px 8px 8px;"
                        >
                          <div
                            class="ant-row c10 card-form-item-view-only"
                          >
                            <div
                              class="ant-col ant-col-8 c11"
                            >
                              Name
                            </div>
                            <div
                              class="ant-col ant-col-16 c12"
                            >
                              Cristiano Ronaldo
                            </div>
                          </div>
                        </div>
                        <div
                          class="ant-col ant-col-24"
                          style="padding: 8px 8px 8px 8px;"
                        >
                          <div
                            class="ant-row c10 card-form-item-view-only"
                          >
                            <div
                              class="ant-col ant-col-8 c11"
                            >
                              Contact Number
                            </div>
                            <div
                              class="ant-col ant-col-16 c12"
                            >
                              7777 7777
                            </div>
                          </div>
                        </div>
                        <div
                          class="ant-col ant-col-24"
                          style="padding: 8px 8px 8px 8px;"
                        >
                          <div
                            class="ant-row c10 card-form-item-view-only"
                          >
                            <div
                              class="ant-col ant-col-8 c11"
                            >
                              Email
                            </div>
                            <div
                              class="ant-col ant-col-16 c12"
                            >
                              manchester_united_are_rubbish@bbc.com
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="css-1ijzjjj-StyledCardSection e5y6tk60"
                    >
                      <span
                        class="ant-typography c8 c9 css-14xdwh8-StyledText-StyledTextCaption emffahp0"
                      >
                        <strong>
                          Ticket Data
                        </strong>
                      </span>
                      <div
                        class="ant-row"
                        style="margin: -8px -8px -8px -8px;"
                      >
                        <div
                          class="ant-col ant-col-24"
                          style="padding: 8px 8px 8px 8px;"
                        >
                          <div
                            class="ant-row c10 card-form-item-view-only"
                          >
                            <div
                              class="ant-col ant-col-8 c11"
                            >
                              Slug
                            </div>
                            <div
                              class="ant-col ant-col-16 c12"
                            >
                              QWERTY
                            </div>
                          </div>
                        </div>
                        <div
                          class="ant-col ant-col-24"
                          style="padding: 8px 8px 8px 8px;"
                        >
                          <div
                            class="ant-row c10 card-form-item-view-only"
                          >
                            <div
                              class="ant-col ant-col-8 c11"
                            >
                              Listing ID
                            </div>
                            <div
                              class="ant-col ant-col-16 c12"
                            >
                              12345
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </section>
</div>
`;
